CDAP-18272: Fix missing binding for InternalAuthenticator

wyzhang opened the pull request:
Otherwise wrangler fails to load data with the following error

1) No implementation for io.cdap.cdap.common.internal.remote.InternalAuthenticator was bound.
  while locating io.cdap.cdap.common.internal.remote.InternalAuthenticator
    for parameter 1 at io.cdap.cdap.common.internal.remote.RemoteClientFactory.<init>(RemoteClientFactory.java:36)
  while locating io.cdap.cdap.common.internal.remote.RemoteClientFactory
    for parameter 0 at io.cdap.cdap.security.store.client.RemoteSecureStore.<init>(RemoteSecureStore.java:58)
  at io.cdap.cdap.security.guice.SecureStoreClientModule.configure(SecureStoreClientModule.java:33)

...

6) No implementation for io.cdap.cdap.common.internal.remote.InternalAuthenticator was bound.
  while locating io.cdap.cdap.common.internal.remote.InternalAuthenticator
    for parameter 1 at io.cdap.cdap.common.internal.remote.RemoteClientFactory.<init>(RemoteClientFactory.java:36)
  while locating io.cdap.cdap.common.internal.remote.RemoteClientFactory
    for parameter 1 at io.cdap.cdap.internal.app.worker.RemoteWorkerPluginFinder.<init>(RemoteWorkerPluginFinder.java:42)
  at io.cdap.cdap.internal.app.worker.SystemAppModule.configure(SystemAppModule.java:70)

dli357 commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/SystemAppTask.java:
```diff
@@ -114,6 +115,7 @@ private Injector createInjector() {
       new MessagingClientModule(),
       new LocalLocationModule(),
       new SecureStoreClientModule(),
+      new AuthenticationContextModules().getProgramContainerModule(cConf),
```
For system apps, we should probably be using getMasterModule() instead

wyzhang replied:
Done

dli357 reviewed (approved):
Thanks!

wyzhang committed:
CDAP-18272: Fix missing binding for InternalAuthenticator

Otherwise wrangler fails to load data with the following error

1) No implementation for io.cdap.cdap.common.internal.remote.InternalAuthenticator was bound.
   while locating io.cdap.cdap.common.internal.remote.InternalAuthenticator
    for parameter 1 at io.cdap.cdap.common.internal.remote.RemoteClientFactory.<init>(RemoteClientFactory.java:36)
   while locating io.cdap.cdap.common.internal.remote.RemoteClientFactory
     for parameter 0 at io.cdap.cdap.security.store.client.RemoteSecureStore.<init>(RemoteSecureStore.java:58)
   at io.cdap.cdap.security.guice.SecureStoreClientModule.configure(SecureStoreClientModule.java:33)

...

6) No implementation for io.cdap.cdap.common.internal.remote.InternalAuthenticator was bound.
  while locating io.cdap.cdap.common.internal.remote.InternalAuthenticator
    for parameter 1 at io.cdap.cdap.common.internal.remote.RemoteClientFactory.<init>(RemoteClientFactory.java:36)
  while locating io.cdap.cdap.common.internal.remote.RemoteClientFactory
    for parameter 1 at io.cdap.cdap.internal.app.worker.RemoteWorkerPluginFinder.<init>(RemoteWorkerPluginFinder.java:42)
  at io.cdap.cdap.internal.app.worker.SystemAppModule.configure(SystemAppModule.java:70)
